logging/logging.go

```diff
@@ -72,25 +96,76 @@ func GetLogLevel() int {
 	return LogDiscard
 }
 
-// CustomizeLogging changes logging format
-func CustomizeLogging(loggingFormat string, serviceName string) {
-	log.SetOutput(os.Stderr)
-	log.SetFormatter(logFormatterFor(loggingFormat, serviceName))
+var (
+	errNoWriter      = errors.New("output writer is not specified")
```

Comment:
why these error variables are private?

Reply:
I looked into @ilammy 's example of keystore builder (https://github.com/cossacklabs/acra/blob/cop/keystore/filesystem/server_keystore.go#L128)

Reply:
I missed it in his PR *(
@ilammy , why you declared err variables private?)

Reply:
Uh...
・・・
\*thinking\*
・・・
I guess that’s just a natural habit of not exporting things that are not needed outside of a module right now and there are no obvious reasons for why they might be needed. I figured that no one is going to write code like

```go
again:
        builder := NewBuilder().ConfigureThis(thing).ConfigureThat(stuff)
        result, err := builder.Build()
        if err != nil {
                if err == ErrAlsoThisNotSpecified {
                        // Oh right, we need to also do this
                        builder.ConfigureAlsoThis(zork)
                        goto again
                }
                // Oh well, no more guesses
                return err
        }
        use(result)
```

so I felt no need to export those errors, because that’s probably the only case where you’d need to have access to those names (you don’t need it to log them, etc.)

Just a natural aversion to expanding the API without a need.

Reply:
On my opinion, it's okay to not export them if they caught internally and don't go out from the package. But I saw that these errors returned from public methods and caller haven't any possible way to handle it by error type. He just can compare it with `nil` or try to convert to string and rely on string representation of error, and it's not okay.

Reply:
> But I saw that these errors returned from public methods and caller haven't any possible way to handle it by error type.

Point taken 👍

I’ll update existing errors in keystore as I go with v2, exporting them whenever they can leave the package boundaries via exported methods.